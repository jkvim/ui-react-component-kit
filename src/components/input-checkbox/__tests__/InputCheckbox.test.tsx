import * as React from 'react';
import { mount } from 'enzyme';
import { InputCheckbox } from '../';
import toJSON from 'enzyme-to-json';

describe('InputCheckbox', () => {
  let wrapper;

  const callback = jest.fn();
  const label = 'Tick this checkbox to confirm that you agree with the following statements';

  beforeEach(() => {
    wrapper = mount(
      <InputCheckbox
        id="testId"
        label={label}
        onChange={callback}
      />
    );
  });

  it('should match snapshot', () => {
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('renders input tag', () => {
    expect(wrapper.find('input')).toHaveLength(1);
  });

  it('should render checkbox unticked', () => {
    const checkbox = wrapper.find('input');
    expect(checkbox.props().checked).toBeFalsy();
  });

  it('renders label', () => {
    expect(wrapper.find('label')).toHaveLength(1);
  });

  it('renders label with passed in label text', () => {
    expect(wrapper.find('label').text()).toBe(label);
  });

  describe('onChange callback', () => {
    test('should call onChange callback when provided', () => {
      expect(callback).not.toBeCalled();
      wrapper.find('input').simulate('change');
      expect(callback).toBeCalled();
    });
  });
});
