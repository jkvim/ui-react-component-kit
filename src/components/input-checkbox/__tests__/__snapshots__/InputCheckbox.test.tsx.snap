// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputCheckbox should match snapshot 1`] = `
<InputCheckbox
  id="testId"
  label="Tick this checkbox to confirm that you agree with the following statements"
  onChange={[MockFunction]}
>
  <styled.div>
    <div
      className="sc-gzVnrw kDUAmj"
    >
      <styled.input
        checked=""
        hasError={false}
        id="testId"
        innerRef={[Function]}
        onChange={[Function]}
        theme={
          Object {
            "color": Object {
              "borderColor": "#333",
              "disableColor": "#acacac",
              "errorColor": "#9f173f",
            },
          }
        }
        type="checkbox"
      >
        <input
          checked=""
          className="sc-ifAKCX bdoYJf"
          id="testId"
          onChange={[Function]}
          type="checkbox"
        />
      </styled.input>
      <styled.label
        hasError={false}
        htmlFor="testId"
        theme={
          Object {
            "color": Object {
              "color": "#333",
              "errorColor": "#9f173f",
            },
          }
        }
      >
        <label
          className="sc-bwzfXH lcbjGL"
          htmlFor="testId"
        >
          Tick this checkbox to confirm that you agree with the following statements
        </label>
      </styled.label>
      <styled.div
        theme={
          Object {
            "color": Object {
              "errorColor": "red",
            },
          }
        }
      >
        <div
          className="sc-iwsKbI ZBWjd"
        />
      </styled.div>
    </div>
  </styled.div>
</InputCheckbox>
`;
